src/physics/reynolds_averaged_navier_stokes.h

```diff
@@ -0,0 +1,252 @@
+#define __REYNOLDS_AVERAGED_NAVIER_STOKES__
+
+#include "model.h"
+#include "navier_stokes.h"
+#include "euler.h"
+
+namespace PHiLiP {
+namespace Physics {
+
+/// Reynolds-Averaged Navier-Stokes equations. Derived from Navier-Stokes for modifying the stress tensor and heat flux, which is derived from PhysicsBase. 
+template <int dim, int nstate, typename real>
+class ReynoldsAveragedNavierStokesBase : public ModelBase <dim, nstate, real>
+{
+public:
+    using thermal_boundary_condition_enum = Parameters::NavierStokesParam::ThermalBoundaryCondition;
+    /// Constructor
+	ReynoldsAveragedNavierStokesBase
+(
+	    const double                                              ref_length,
+        const double                                              gamma_gas,
+        const double                                              mach_inf,
+        const double                                              angle_of_attack,
+        const double                                              side_slip_angle,
+        const double                                              prandtl_number,
+        const double                                              reynolds_number_inf,
+        const double                                              turbulent_prandtl_number,
+        const double                                              isothermal_wall_temperature = 1.0,
+        const thermal_boundary_condition_enum                     thermal_boundary_condition_type = thermal_boundary_condition_enum::adiabatic,
+        std::shared_ptr< ManufacturedSolutionFunction<dim,real> > manufactured_solution_function = nullptr);
+
+    /// Destructor
+    ~ReynoldsAveragedNavierStokesBase() {};
+
+    /// Turbulent Prandtl number
+    const double turbulent_prandtl_number;
+
+    /// Pointer to Navier-Stokes physics object
+    std::unique_ptr< NavierStokes<dim,dim+2,real> > navier_stokes_physics;
+
+    /// Additional convective flux of RANS + convective flux of turbulence model
+    std::array<dealii::Tensor<1,dim,real>,nstate> convective_flux (
+        const std::array<real,nstate> &conservative_soln) const;
+
+    /// Additional viscous flux of RANS + viscous flux of turbulence model
+    std::array<dealii::Tensor<1,dim,real>,nstate> dissipative_flux (
+        const std::array<real,nstate> &conservative_soln,
+        const std::array<dealii::Tensor<1,dim,real>,nstate> &solution_gradient,
+        const dealii::types::global_dof_index cell_index) const;
+
+    std::array<dealii::Tensor<1,dim,real>,nstate> convective_numerical_split_flux (
+        const std::array<real,nstate> &conservative_soln1,
+        const std::array<real,nstate> &conservative_soln2) const;
+
+    /// Convective eigenvalues
+    std::array<real,nstate> convective_eigenvalues (
+        const std::array<real,nstate> &/*conservative_soln*/,
+        const dealii::Tensor<1,dim,real> &/*normal*/) const;
+
+    /// Maximum convective eigenvalue used in Lax-Friedrichs
```

Comment:
update according to other comments regarding this

Reply:
Updated the comments based on the current changes